client/components/social-buttons/apple.js

```diff
@@ -0,0 +1,252 @@
+import { recordTracksEventWithClientId as recordTracksEvent } from 'state/analytics/actions';
+import { isFormDisabled } from 'state/login/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class AppleLoginButton extends Component {
+	static propTypes = {
+		isFormDisabled: PropTypes.bool,
+		clientId: PropTypes.string.isRequired,
+		scope: PropTypes.string,
+		fetchBasicProfile: PropTypes.bool,
+		uxMode: PropTypes.string,
+		recordTracksEvent: PropTypes.func.isRequired,
+		responseHandler: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+		onClick: PropTypes.func,
+	};
+
+	static defaultProps = {
+		scope: 'https://www.googleapis.com/auth/userinfo.profile',
+		fetchBasicProfile: true,
+		onClick: noop,
+	};
+
+	state = {
+		error: '',
+		showError: false,
+		errorRef: null,
+		isDisabled: true,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.initialized = null;
+
+		this.handleClick = this.handleClick.bind( this );
+		this.showError = this.showError.bind( this );
+		this.hideError = this.hideError.bind( this );
+	}
+
+	componentDidMount() {
+		this.initialize();
+	}
+
+	async loadDependency() {
+		if ( ! window.appleLoginApi ) {
+			await loadScript(
+				'https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js'
+			);
+		}
+
+		return window.appleLoginApi;
+	}
+
+	initialize() {
+		if ( this.initialized ) {
+			return this.initialized;
+		}
+
+		this.setState( { error: '' } );
+
+		const { translate } = this.props;
+
+		this.initialized = this.loadDependency()
+			.then( appleLoginApi =>
+				appleLoginApi.auth.init( {
+					clientId: '[CLIENT_ID]',
+					scope: '[SCOPES]',
+					redirectURI: '[REDIRECT_URI]',
+					state: '[STATE]',
+				} )
+			)
+			.then( gapi =>
+				gapi.client
+					.init( {
+						fetch_basic_profile: this.props.fetchBasicProfile,
+						ux_mode: this.props.uxMode,
+						redirect_uri: this.props.redirectUri,
+					} )
+					.then( () =>
+						gapi.auth2
+							.init( {
+								client_id: this.props.clientId,
+								scope: this.props.scope,
+							} )
+							.then( () => {
+								this.setState( { isDisabled: false } );
+
+								const googleAuth = gapi.auth2.getAuthInstance();
+								const currentUser = googleAuth.currentUser.get();
+
+								// handle social authentication response from a redirect-based oauth2 flow
+								if ( currentUser && this.props.uxMode === 'redirect' ) {
+									this.props.responseHandler( currentUser, false );
+								}
+
+								return gapi; // don't try to return googleAuth here, it's a thenable but not a valid promise
+							} )
+					)
+			)
+			.catch( error => {
+				this.initialized = null;
+
+				if ( 'idpiframe_initialization_failed' === error.error ) {
+					// This error is caused by 3rd party cookies being blocked.
+					this.setState( {
+						error: translate(
+							'Please enable "third-party cookies" to connect your Apple account.'
```

Comment:
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Please enable "third-party cookies" to connect your Google account.' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).